src/main/java/com/yahoo/identity/services/credential/CredentialImpl.java

```diff
@@ -0,0 +1,118 @@
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.yahoo.identity.services.key.KeyServiceImpl;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.BadRequestException;
+
+public class CredentialImpl implements Credential {
+
+    private Instant issueTime;
+    private Instant expireTime;
+    private String subject;
+    private KeyServiceImpl keyServiceImpl = new KeyServiceImpl();
+
+    @Override
+    @Nonnull
+    public Instant getIssueTime() {
+        return this.issueTime;
+    }
+
+    public void setIssueTime(@Nonnull Instant issueTime) {
+        this.issueTime = issueTime;
+    }
+
+    @Override
+    @Nonnull
+    public Instant getExpireTime() {
+        return this.expireTime;
+    }
+
+    public void setExpireTime(@Nonnull Instant expireTime) {
+        this.expireTime = expireTime;
+    }
+
+    @Override
+    @Nonnull
+    public String getSubject() {
+        return this.subject;
+    }
+
+    public void setSubject(@Nonnull String subject) {
+        this.subject = subject;
+    }
+
+    @Override
+    @Nonnull
+    public String toString() {
+        try {
+            Algorithm algorithm = Algorithm.HMAC256(keyServiceImpl.getSecret());
+            String token = JWT.create()
+                .withExpiresAt(Date.from(expireTime))
+                .withIssuedAt(Date.from(issueTime))
+                .withSubject(subject)
+                .sign(algorithm);
+            return token;
+        } catch (JWTCreationException exception) {
+            throw new BadRequestException("JWT creation not succeeded.");
+        }
+    }
+
+    public String fromString(@Nonnull String credStr) {
+        try {
+            Algorithm algorithm = Algorithm.HMAC256(keyServiceImpl.getSecret());
+
+            JWTVerifier verifier = JWT.require(algorithm)
+                .acceptLeeway(1)   // 1 sec for nbf and iat
+                .acceptExpiresAt(5)   // 5 secs for exp
+                .build();
+
+            DecodedJWT jwt = verifier.verify(credStr);
+
+            setSubject(jwt.getSubject());
+            setIssueTime(jwt.getIssuedAt().toInstant());
+            setExpireTime(jwt.getExpiresAt().toInstant());
+
+        } catch (JWTVerificationException e) {
+            throw new BadRequestException("JWT verification not succeeded.");
+        }
+        return getSubject();
+    }
+
+    public String fromStringLimit(@Nonnull String credStr) {
+        try {
+            Algorithm algorithm = Algorithm.HMAC256(keyServiceImpl.getSecret());
+
+            JWTVerifier verifier = JWT.require(algorithm)
+                .acceptLeeway(1)   // 1 sec for nbf and iat
+                .acceptExpiresAt(5)   // 5 secs for exp
+                .build();
+
+            DecodedJWT jwt = verifier.verify(credStr);
+
+            setSubject(jwt.getSubject());
+            System.out.println(getSubject());
```

Comment:
Don't print to stdout in the code.
If you want to log, use slf4j + log4j2 (not log4j, which is older, or you can use logback if you want) .